compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/cfa/FirPropertyInitializationChecker.kt

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.fir.analysis.cfa
+
+import kotlinx.collections.immutable.PersistentMap
+import kotlinx.collections.immutable.persistentMapOf
+import org.jetbrains.kotlin.contracts.description.EventOccurrencesRange
+import org.jetbrains.kotlin.fir.analysis.diagnostics.DiagnosticReporter
+import org.jetbrains.kotlin.fir.references.FirResolvedNamedReference
+import org.jetbrains.kotlin.fir.resolve.dfa.cfg.*
+import org.jetbrains.kotlin.fir.symbols.impl.FirPropertySymbol
+
+abstract class FirPropertyInitializationChecker {
```

Comment:
`AbstractFirPropertyInitializationChecker` will be better name

Reply:
Also there should be an abstract checker, like
```
abstract class FirControlFlowChecker {
    abstract fun analyze(graph: ControlFlowGraph, reporter: DiagnosticReporter)
}
```
so `AbstractFirPropertyInitializationChecker` will inherit it

Reply:
There are two notes that contradicts each other a bit. I think you should introduce `FirControlFlowChecker` with an API given above ^ and inherit `AbstractFirPropertyInitializationChecker` from it.

Reply:
The `FirControlFlowChecker ` have been exist, but I removed it (see comment below). I was misunderstood this?